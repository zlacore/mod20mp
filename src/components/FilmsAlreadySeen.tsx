import type React from 'react';
import type Film from '../utils/interfaces/Film.interface';
import FilmCard from './FilmCard';

interface seenFilmProps {
  alreadyWatchedFilms: Film[];
  removeFromStorage:
    | ((
        e: React.MouseEvent<SVGSVGElement, MouseEvent>,
        currentlyOnWatchList: boolean | null | undefined,
        currentlyOnSeenItList: boolean | null | undefined,
        title: string | null
      ) => void)
    | null;
}

const FilmsAlreadySeen = ({
  alreadyWatchedFilms,
  removeFromStorage,
}: seenFilmProps) => {
  return (
    <ul>
      {alreadyWatchedFilms.map((film) => (
        <FilmCard
          currentFilm={film}
          key={film.Title}
          onSeenItList={true}
          removeFromStorage={removeFromStorage}
        />
      ))}
    </ul>
  );
};

export default FilmsAlreadySeen;
