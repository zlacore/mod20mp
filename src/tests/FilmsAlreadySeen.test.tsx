import pretty from 'pretty';
import { render } from '@testing-library/react';
import FilmsAlreadySeen from '../components/FilmsAlreadySeen';
import type Film from '../utils/interfaces/Film.interface';

const films: Film[] = [{
  Title: 'Gamera, the Guardian of the Universe',
  Director: 'ShÃ»suke Kaneko',
  Actors: 'Tsuyoshi Ihara, Akira Onodera, Shinobu Nakayama',
  Released: '16 Apr 1997',
  Poster: 'https://m.media-amazon.com/images/M/MV5BNTFkNzAxYTUtZGE4ZC00NjBkLWE3ZDAtYjE4NzQ0MDBhZWQwXkEyXkFqcGdeQXVyNjQ2MzU1NzQ@._V1_SX300.jpg',
  Genre: 'Action, Drama, Fantasy',
  Plot: 'An ornithologist investigates reports of a monstrous new species of bird just as a teenage girl is gifted an amulet found on mysterious atoll. As the creatures begin to attack, an ancient guardian with a bond to the girl emerges.',
},
{
  Title:"Little Nemo: Adventures in Slumberland",
  Director: 'Masami Hata, Masanori Hata, William T. Hurtz',
  Actors: 'Gabriel Damon, Mickey Rooney, Rene Auberjonois',
  Released: '21 Aug 1992',
  Poster: 'https://m.media-amazon.com/images/M/MV5BMmNkM2MzNDgtNzk3Ny00ZTZmLTg4MDMtODc5M2Y4ZGQ4MGMyXkEyXkFqcGdeQXVyNjMwMjk0MTQ@._V1_SX300.jpg',
  Genre: 'Animation, Adventure, Comedy',
  Plot: 'Each night, young Nemo goes to Slumberland and has adventures and befriends the king of Slumberland, Morpheus. But one night Nemo discovers Nightmare Land, and the evil nightmare king, throwing Slumberland and Nemo himself into danger.',
},
{
  Title: 'FernGully: The Last Rainforest',
  Director: 'Bill Kroyer',
  Actors: 'Samantha Mathis, Christian Slater, Robin Williams',
  Released: '19 Jun 1992',
  Poster: 'https://m.media-amazon.com/images/M/MV5BNzNiNTliMjItMmY0ZS00YTMzLTllYTQtMzE2ZDM1YTgwZjlmL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyMTEwMTkwOTI@._V1_SX300.jpg',
  Genre: 'Animation, Adventure, Family',
  Plot: 'The magical inhabitants of a rainforest fight to save their home, which is threatened by logging and a polluting force of destruction called Hexxus.',
}];

describe('FilmsAlreadySeen', () => {

  it('should match snapshot', () => {
    render(<FilmsAlreadySeen alreadyWatchedFilms={films} removeFromStorage={null}/>);

    const filmContainer = document.querySelector('.filmCard') as HTMLElement;

    if (filmContainer) {
      expect(pretty(filmContainer.innerHTML)).toMatchSnapshot();
    }
  })
  
});
