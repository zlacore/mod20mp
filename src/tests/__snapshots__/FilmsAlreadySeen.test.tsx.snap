// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilmsAlreadySeen > should match snapshot 1`] = `
"<figure><img alt="Gamera, the Guardian of the Universe" src="https://m.media-amazon.com/images/M/MV5BNTFkNzAxYTUtZGE4ZC00NjBkLWE3ZDAtYjE4NzQ0MDBhZWQwXkEyXkFqcGdeQXVyNjQ2MzU1NzQ@._V1_SX300.jpg"></figure>
<article class="details">
  <h2>Gamera, the Guardian of the Universe</h2>
  <p><strong>Directed By:</strong> Sh√ªsuke Kaneko</p>
  <p><strong>Starring:</strong> Tsuyoshi Ihara, Akira Onodera, Shinobu Nakayama</p>
  <p><strong>Released:</strong> 16 Apr 1997</p>
  <p><strong>Genre:</strong> Action, Drama, Fantasy</p>
</article>
<article class="plot">
  <p><strong>Plot:</strong> An ornithologist investigates reports of a monstrous new species of bird just as a teenage girl is gifted an amulet found on mysterious atoll. As the creatures begin to attack, an ancient guardian with a bond to the girl emerges.</p>
</article>
<aside class="icons"><svg stroke="currentColor" fill="currentColor" stroke-width="0" version="1.1" viewBox="0 0 16 16" style="font-size: 40px; cursor: pointer;" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
    <path d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"></path>
  </svg></aside>"
`;
